### PR TITLE
fix: display "Resend" for expired messages

### LINKED ISSUE #7769: Resend button is not shown for expired messages

### PR DESCRIPTION
**Pull request by richard-ramos**
Fixes: #7769

### CODE REVIEW

**Review by alaibe — APPROVED** [review]

File: src/app/modules/shared_models/message_model.nim [hunk 1]
```diff
@@ -356,6 +356,7 @@ QtObject:
 
   proc setOutgoingStatus(self: Model, messageId: string, status: string) =
     let ind = self.findIndexForMessageId(messageId)
+    echo "IND IS", ind
```
Comment (caybro) [inline src/app/modules/shared_models/message_model.nim:359]: Leftover debug?

**Review by caybro — APPROVED** [review]

File: src/app/modules/shared_models/message_model.nim (hunk 1 above)
Comment (richard-ramos) [reply src/app/modules/shared_models/message_model.nim:359]: oops

**Review by MishkaRogachev — APPROVED** [review]

**Comment by elina2015** [comment]
@richard-ramos hey, it seems the build is not correct - after launching it's closing without any notification. Could you kindly rebuild?

**Review by elina2015 — CHANGES_REQUESTED** [review]
The build doesn't work - macOS Monterey

**Comment by elina2015** [comment]
@richard-ramos hey, I sent few messages in the test chat after disconnecting (the 'Disconnected' banner appeared) and after sending the messages I reconnected back. But the Resend option is not shown:
image
At the same time, my  offline messages are nor shown for other users (it's a proof that they weren't sent):
image
Could you  kindly check it?

**Comment by richard-ramos** [comment]
Looking into this issue, i saw something interesting that would require some UI work for the message items
1. Login with WiFi, enter a channel
2. Make sure the last message written in the channel is not yours
(this can be done by joining a new random name channel) 
3. Disconnect from WiFi and wait for disconnected banner to popup
4. Write a message, The Resend label should appear

However, if you have written a message before, the resend button will not appear, since the message gets grouped with the previous message header (containing username and time).

IMO he resend button should not appear only besides the username, but appear in each message individually. i.e.: in this image these 3 messages were sent while i was disconnected, yet the resend button is only available for the first message
image

cc: @alexandraB99

**Comment by elina2015** [comment]
@richard-ramos yes, it works as you've described! So, I will just open a new issue about the Resend button doesn't appear if there is some messages from me before disconnecting.

**Review by elina2015 — APPROVED** [review]
Tested and approved

**Pull request merged**